### PR TITLE
[refactor][broker] Arrange cleanup operation for topic creation

### PR DESCRIPTION
**Pull request by Shawyeok**
### Motivation

Original from https://github.com/apache/pulsar/pull/15139#issuecomment-1098692062.

If create topic fails, many code branch handle cleanup operation for `topics` map like `pulsar.getExecutor().execute(() -> topics.remove(topic, topicFuture));`, handle it in separate thread to avoid deadlock, and some code branch doesn't. IMO, it should be handled properly in just one place.

### Modifications

1. handle cleanup operation before return future result
2. delete other cleanup operation except one with specific reason https://github.com/Shawyeok/pulsar/blob/78bbbe66d5abf66fc0c6ce15624107227a17a360/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java#L1190

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [x] `no-need-doc` 
Just refactor work

### CODE REVIEW

**Review by lhotari — APPROVED** [review]
LGTM. Useful refactoring. Good work @Shawyeok .

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java [hunk 1]
```diff
@@ -1432,9 +1434,7 @@ public void openLedgerComplete(ManagedLedger ledger, Object ctx) {
                                             if (topicFuture.isCompletedExceptionally()) {
                                                 log.warn("{} future is already completed with failure {}, closing the"
                                                         + " topic", topic, FutureUtil.getException(topicFuture));
-                                                persistentTopic.stopReplProducers().whenComplete((v, exception) -> {
```
Comment (poorbarcode) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java:1435]: In original implements, It ensures that `persistentTopic.stopReplProducers()` executes before `topics.remove(...)`, this ensures that "there are no multiple threads change one topic"

After this change, this advantage was deleted.

Comment (poorbarcode) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java:1435]: In original implements, It ensures that `persistentTopic.stopReplProducers()` executes before `topics.remove(...)`, this ensures that "there are no multiple threads change one topic"

After this change, this advantage was deleted.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java [hunk 2]
```diff
@@ -1432,9 +1434,7 @@ public void openLedgerComplete(ManagedLedger ledger, Object ctx) {
...
-                                                    topics.remove(topic, topicFuture);
-                                                });
+                                                persistentTopic.stopReplProducers();
```
Comment (poorbarcode) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java:1437]: In original implements, It ensures that `persistentTopic.stopReplProducers()` executes before `topics.remove(...)`, this ensures that "there are no multiple threads change one topic"

After this change, this advantage was deleted.

**Comment by tisonkun** [comment]
Closed as stale and conflict. Please rebase and resubmit the patch if it's still relevant.

**Pull request closed without merging**